phonotype.lua

```diff
@@ -277,6 +282,12 @@ function keyboard.code(key, value)
       if edit_row > model:script_size(editing_script) then
         edit_row = model:script_size(editing_script) + 1
       else
+        if keyboard.alt() and model:get(editing_script, edit_row + 1) ~= "" then
+          local storedLine = model:get(editing_script, edit_row + 1)
+          model:replace(editing_script, edit_row + 1, editing)
```

Comment:
So the problem here is that the source of truth is supercollider, and supercollider takes some time to parse the new line and commit it to the server — and we don't update the UI until it does. 

I think the simplest solution, which I have been considering in general, is to embargo all further changes until the change in progress has been committed to the server-side model, but that will make moving lines around in this manner quite sluggish. 

Another option is to hold the intent of where to move the line until the user releases ctrl or alt, and only _then_ attempt to tell the engine about the move (using a remove and an insert operation, instead of a replace — later to be changed to a "move" operation that I implement directly in sc). That leaves the problem of representing the user's intent in the cached version of the lines in the meantime, and detecting when the user releases the meta key to know when to commit the change.

Dang, a can of worms indeed.